### PR TITLE
OSS-694: Validate input parameters

### PR DESCRIPTION
**Pull request by shestakovg**
…thods

### CODE REVIEW

**Review by parkhomenko — DISMISSED** [review]

File: FaunaDB.Client/Client/DefaultClientIO.cs [hunk 1]
```diff
@@ -34,6 +34,22 @@ class DefaultClientIO : IClientIO
...
+                throw new ArgumentNullException(nameof(authHeader));
+            }
+            if (endpoint == null)
+            {
+                throw new ArgumentNullException(nameof(endpoint));
+            }
+            if (lastSeen == null)
+            {
+                throw new ArgumentNullException(nameof(lastSeen));
+            }
```
Comment (evbruno) [inline FaunaDB.Client/Client/DefaultClientIO.cs:52]: nice sanitizing !!

Comment (evbruno) [inline FaunaDB.Client/Client/DefaultClientIO.cs:52]: can we create an auxiliary method so we can "compact" those calls, something like:
```          
            AssertNotNull( client );
            AssertNotNull( authHeader );
            AssertNotNull( endpoint );
            AssertNotNull( lastSeen );
```

File: FaunaDB.Client/Client/FaunaClient.cs [hunk 2]
```diff
@@ -46,6 +46,10 @@ public class FaunaClient
         /// </summary>
         public FaunaClient(IClientIO clientIO)
         {
+            if (clientIO == null)
+            {
+                throw new ArgumentNullException(nameof(clientIO));
+            }
             this.clientIO = clientIO;
```
Comment (evbruno) [inline FaunaDB.Client/Client/FaunaClient.cs:53]: same here

**Review by evbruno — CHANGES_REQUESTED** [review]
just a minor comment; its not blocking the PR and you can dismiss if you like too

File: FaunaDB.Client/Client/DefaultClientIO.cs (hunk 1 above)
Comment (shestakovg) [reply FaunaDB.Client/Client/DefaultClientIO.cs:52]: Really it's good idea. Will do

File: FaunaDB.Client/Errors/NotNullValidator.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace FaunaDB.Errors
+{
+    static class NotNullValidator
+    {
+        public static void AssertNotNull(this object value, string valueName)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(valueName);
```
Comment (evbruno) [inline FaunaDB.Client/Errors/NotNullValidator.cs:11]: can we use `nameof` here so the callers are even more simple?

```suggestion
                throw new ArgumentNullException(nameof(valueName));
```

**Review by evbruno — CHANGES_REQUESTED** [review]
I think it would be nice to add some tests (at least for `FaunaClient`)

**Comment by evbruno** [comment]
I'm still wondering if this is better accomplished w/ the extension method proposed (as there are no base types, only `object`) or a class method (static). Any reasons for that? (not a c# expert, though)

**Comment by evbruno** [comment]
miss clicked on`Close` instead of `Comment`, my bad 🤦

**Review by evbruno — CHANGES_REQUESTED** [review]
there are some comments/questions on the PR, but the main change that I would like to see is (at least) one test for the inputs; if so, the `FaunaClient` constructor should suffice 😉

**Review by evbruno — DISMISSED** [review]
🚢 -it

**Review by parkhomenko — APPROVED** [review]

**Review by parkhomenko — APPROVED** [review]

**Review by vadimLF — APPROVED** [review]
LGTM. Just confusing with extension method name which looks like some xUnit method (maybe checkNotNull maybe ?)

**Pull request merged**